index.js

```diff
@@ -44,6 +44,14 @@ var isArray = Array.isArray;
 // interdependence between different functions.
 var conjure;
 
+// a special hasOwnProperty function so we can override the built-in one
+
+var hasOwnProperty = Object.prototype.hasOwnProperty;
+
+function hop(obj, propName) {
```

Comment:
I'm confused why not:
```
// Keep it on the prototype underscored incase we need it later
Object.prototype._hasOwnProperty = Object.prototype.hasOwnProperty;
Object.prototype.hasOwnProperty = function (propName) {
    return hasOwnProperty.call(this, propName);
};
```

and later use `target.hasOwnProperty(propertyName);`???